crates/bevy_render/src/camera/visible_entities.rs

```diff
@@ -24,22 +24,208 @@ impl VisibleEntities {
     }
 }
 
+type LayerMask = u32;
+
+/// An identifier for a rendering layer.
+pub type Layer = u8;
+
+/// Describes which rendering layers an entity belongs to.
+///
+/// Cameras with this component will only render entities with intersecting
+/// layers.
+///
+/// There are 32 layers numbered `0` - [`TOTAL_LAYERS`]. Entities may belong to one or more
+/// layers, or no layer at all.
+///
+/// The [`Default`] instance of `RenderLayers` contains layer `0`, the first layer.
+///
+/// An entity with this component without any layers is invisible.
+///
+/// Entities without this component belong to layer `0`.
+#[derive(Copy, Clone, Reflect, PartialEq, Eq, PartialOrd, Ord)]
+#[reflect(Component)]
+pub struct RenderLayers(LayerMask);
+
+impl std::fmt::Debug for RenderLayers {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("RenderLayers")
+            .field(&self.layers_iter().collect::<Vec<_>>())
+            .finish()
```

Comment:
```suggestion
        self.layers_iter()
            .fold(f.debug_tuple(), |t, layer| t.field(layer))
            .finish()
```
Currently, this will create a single vec field instead of a field for each layer

This suggests that this code could do with a test.

Reply:
Currently there is a test for the debug output. It is rendering as `"RenderLayers([0, 1, 2, 3])"`.

Reply:
Ah, I see. I was thinking the intention was for `"RenderLayers(0, 1, 2, 3)"`, which to me seems just as reasonable. Both are fine though.